src/libsyntax/parse/parser.rs

```diff
@@ -5391,14 +5391,10 @@ impl<'a> Parser<'a> {
             }
             let mut err = self.diagnostic().struct_span_err(id_sp,
                 "cannot declare a new module at this location");
-            let this_module = match self.directory.path.file_name() {
-                Some(file_name) => file_name.to_str().unwrap().to_owned(),
-                None => self.root_module_name.as_ref().unwrap().clone(),
-            };
             err.span_note(id_sp,
                           &format!("maybe move this module `{0}` to its own directory \
                                      via `{0}/mod.rs`",
-                                    this_module));
+                                    paths.name));
```

Comment:
Hum, the issue where I got this didn't reflect such a case. I'll need to take a different look. Thanks for the help!